### PR TITLE
fix #1834

### LINKED ISSUE #1834: scroll.js: replaceState() is not supported in IE9
### Version
3.0.1

### What is actually happening?
IE9 throw the following error: `Object doesn't support this property or method "replaceState"`.
This was introduced here:
commit: https://github.com/vuejs/vue-router/commit/c166822cdc6cadd9f3c01085b877c86d74362b95
related issue: https://github.com/vuejs/vue-router/issues/1585

### PR DESCRIPTION
**Pull request by JounQin**
fix issue on browsers which do not support history API

### CODE REVIEW

**Comment by callumacrae** [comment]
What happens in browsers that don't support history API? Are they just stuck on the wrong URL now?

**Comment by JounQin** [comment]
@callumacrae As the origin issue mentions, it throws an error on IE9: `Object doesn't support this property or method "replaceState"`.

File: src/history/html5.js [hunk 1]
```diff
@@ -5,15 +5,16 @@ import { History } from './base'
...
 
 export class HTML5History extends History {
   constructor (router: Router, base: ?string) {
     super(router, base)
 
     const expectScroll = router.options.scrollBehavior
+    const supportsScroll = supportsPushState && expectScroll
 
-    if (expectScroll) {
+    if (supportsScroll) {
```
Comment (yyx990803) [inline src/history/html5.js:17]: should also change to `supportScroll` on line 33

**Review by yyx990803 — CHANGES_REQUESTED** [review]

File: src/history/html5.js (hunk 1 above)
Comment (JounQin) [reply src/history/html5.js:17]: Done.

**Pull request merged**